elfi/methods/post_processing.py

```diff
@@ -0,0 +1,233 @@
+        parameter_names : list[str]
+          a list of parameter names
+        summary_names : list[str]
+          a list of names for the summary nodes
+        """
+        self._X = self._input_variables(model, result, summary_names)
+        self._result = result
+        self._parameter_names = parameter_names
+
+        for name in parameter_names:
+            self.regression_models.append(self._fit1(self._X, result.outputs[name]))
+
+        self._fitted = True
+
+    def _fit1(self, X, y):
+        return self._regression_model(**self._model_kwargs).fit(X, y)
+
+    def adjust(self):
+        """Adjust the posterior.
+
+        Returns
+        -------
+          a Result object containing the adjusted posterior
+        """
+        outputs = {}
+        for (i, name) in enumerate(self.parameter_names):
+            theta_i = self.result.outputs[name]
+            adjusted = self._adjust(theta_i, self.regression_models[i])
+            outputs[name] = adjusted
+
+        res = results.Result(method_name=self._name, outputs=outputs,
+                             parameter_names=self._parameter_names)
+        return res
+
+    def _adjust(self, theta_i, regression_model):
+        """Adjust a single parameter using a fitted regression model.
+
+        Parameters
+        ----------
+        theta_i : np.ndarray
+          a vector of parameter values to adjust
+        regression_model
+          a fitted regression model
+
+        Returns
+        -------
+        adjusted_theta_i : np.ndarray
+          an adjusted version of the parameter values
+        """
+        raise NotImplementedError
+
+    def _input_variables(self, model, result, summary_names):
+        """Construct a matrix of regressors.
+
+        Parameters
+        ----------
+        model : elfi.ElfiModel
+          the inference model
+        result
+          a result object from an ABC algorithm
+        summary_names : list[str]
+          names of the summary nodes
+
+        Returns
+        -------
+        X
+          a numpy array of regressors
+        """
+        raise NotImplementedError
+
+
+class LinearAdjustment(RegressionAdjustment):
+    """Regression adjustment using a local linear model."""
+    _regression_model = LinearRegression
+    _name = 'LinearAdjustment'
+
+    def __init__(self, **kwargs):
+        super(LinearAdjustment, self).__init__(**kwargs)
+
+    def _adjust(self, theta_i, regression_model):
+        b = regression_model.coef_
+        return theta_i - self.X.dot(b)
+        
+    def _input_variables(self, model, result, summary_names):
+        """Regress on the differences to the observed summaries."""
+        observed_summaries = np.stack([model[s].observed for s in summary_names], axis=1)
+        summaries = np.stack([result.outputs[name] for name in summary_names], axis=1)
+        return summaries - observed_summaries
+
+
+def adjust_posterior(model, result, parameter_names, summary_names, adjustment=None):
```

Comment:
I would suggest the following:
- Add a copy of the model to the result (so that model does not need to be passed). I think this would be nicer.
- Parameter names should be already in the result (this should be made optional) and at least are in the model
- Summary names can be automatically determined by investigating the types of the nodes (query their state and check the class) so make this optional as well.